### PR TITLE
Add support for scalar parameter types and return types

### PR DESCRIPTION
**Pull request by everzet**
Finalisation of #200

### CODE REVIEW

**Comment by everzet** [comment]
@trowski thank you very much for contribution!

File: spec/Prophecy/Prophecy/MethodProphecySpec.php [hunk 1]
```diff
@@ -4,6 +4,11 @@
 
 use PhpSpec\ObjectBehavior;
 
+class ClassWithFinalMethod
```
Comment (stof) [inline spec/Prophecy/Prophecy/MethodProphecySpec.php]: are you sure about this change ? I thought PhpSpec would not run files properly if the spec is not the first class in it

File: spec/Prophecy/Doubler/Generator/ClassMirrorSpec.php [hunk 2]
```diff
@@ -86,6 +92,10 @@ function it_changes_argument_names_if_they_are_varying($class, $method, $paramet
         $method->isStatic()->willReturn(false);
         $method->returnsReference()->willReturn(false);
 
+        if (version_compare(PHP_VERSION, '7.0', '>=')) {
```
Comment (lisachenko) [inline spec/Prophecy/Doubler/Generator/ClassMirrorSpec.php]: I think, that checking of PHP_VERSION_ID can be faster and more transparent.

**Comment by trowski** [comment]
@everzet You're welcome. Sorry that there was some things to clean up, that wasn't my intention. `MethodNode::hasReturnType()` should have always returned `false` in any tests that would be run under 5.x and that there would be no need for all the version comparisons. Not sure where that went wrong...

**Pull request merged**